dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/TokenDAO.java

```diff
@@ -112,4 +118,22 @@ public List<Token> findAllGoogleTokens() {
     public List<Token> findAllBitBucketTokens() {
         return list(namedTypedQuery("io.dockstore.webservice.core.Token.findAllBitbucketTokens"));
     }
+
+    public Token createDockstoreToken(long userId, String username) {
```

Comment:
Not sure, but I think this could be static and final

Reply:
Probably missed, not a big deal but simplifies mental load

Reply:
Not changed on purpose.  `createDockstoreToken` is just another method of `TokenDAO`, and regarding `static`/`final`, has the same signature as other similar methods.  I'm in the camp that `static`/`final` should only be used on methods when there's a compelling architectural or performance justification (like avoiding virtual function overhead, explicitly disabling overrides because they don't make sense or would be dangerous, a function in a helper class, etc), and there's no such justification here.  `createDockstoreToken` is simply a method of `TokenDAO`, and someday we might want to reference non-static fields or override it, and making it `static`/`final` forecloses on those options and makes the codebase less flexible.

Reply:
FWIW, I think we've been down this discussion before with Charles, but this is confusing because it's contrary to what one would normally expect in Java. The point is to simplify understanding of the method by making it clear that the method as-is doesn't use instance variables. If we want it to in the future, it becomes an explicit choice to remove static/final

e.g. 

> One rule-of-thumb: ask yourself "Does it make sense to call this method, even if no object has been constructed yet?" If so, it should definitely be static.
https://stackoverflow.com/questions/2671496/when-to-use-static-methods

This looks like a static factory method to me https://www.baeldung.com/java-constructors-vs-static-factory-method

Reply:
nvm, `persist` two calls can't be static